### PR TITLE
Add more log configs

### PR DESCRIPTION
**Pull request by pureklkl**
**Description:** 
Adding a feature, expose more configs for the collector zap logger.

```
// DisableCaller stops annotating logs with the calling function's file
// name and line number. By default, all logs are annotated.
DisableCaller bool `mapstructure:"disableCaller"`

// DisableStacktrace completely disables automatic stacktrace capturing. By
// default, stacktraces are captured for WarnLevel and above logs in
// development and ErrorLevel and above in production.
DisableStacktrace bool `mapstructure:"disableStacktrace"`

// OutputPaths is a list of URLs or file paths to write logging output to.
// See Open for details.
OutputPaths []string `mapstructure:"outputPaths"`

// ErrorOutputPaths is a list of URLs to write internal logger errors to.
// The default is standard error.
//
// Note that this setting only affects internal errors; for sample code that
// sends error-level logs to a different location from info- and debug-level
// logs, see the package-level AdvancedConfiguration example.
ErrorOutputPaths []string `mapstructure:"errorOutputPaths"`

// InitialFields is a collection of fields to add to the root logger.
InitialFields map[string]interface{} `mapstructure:"initialFields"``

```

**Link to tracking Issue:** #1048 

**Testing:** enhance the current marshaler test to include the new configs, tested manually.

### CODE REVIEW

File: config/service.go [hunk 1]
```diff
@@ -55,6 +55,30 @@ type ServiceTelemetryLogs struct {
 	// Encoding sets the logger's encoding.
 	// Valid values are "json" and "console".
 	Encoding string `mapstructure:"encoding"`
+
+	// DisableCaller stops annotating logs with the calling function's file
+	// name and line number. By default, all logs are annotated.
+	DisableCaller bool `mapstructure:"disableCaller"`
```
Comment (bogdandrutu) [inline config/service.go:61]: I like the comment of explaining the default behavior, but as engineer I would like to see the default "value" as well. Something like:

```
(default = true)
```

Or however you want.

File: config/service.go [hunk 2]
```diff
@@ -55,6 +55,30 @@ type ServiceTelemetryLogs struct {
...
+
+	// DisableStacktrace completely disables automatic stacktrace capturing. By
+	// default, stacktraces are captured for WarnLevel and above logs in
+	// development and ErrorLevel and above in production.
+	DisableStacktrace bool `mapstructure:"disableStacktrace"`
+
+	// OutputPaths is a list of URLs or file paths to write logging output to.
+	// See Open for details.
```
Comment (bogdandrutu) [inline config/service.go:69]: See what? I think this comment is not clear for our users.

File: config/service.go [hunk 3]
```diff
@@ -55,6 +55,30 @@ type ServiceTelemetryLogs struct {
...
+	OutputPaths []string `mapstructure:"outputPaths"`
+
+	// ErrorOutputPaths is a list of URLs to write internal logger errors to.
```
Comment (bogdandrutu) [inline config/service.go:72]: Are these "URLs"?

File: config/service.go [hunk 4]
```diff
@@ -55,6 +55,30 @@ type ServiceTelemetryLogs struct {
...
+	// The default is standard error.
+	//
+	// Note that this setting only affects internal errors; for sample code that
+	// sends error-level logs to a different location from info- and debug-level
+	// logs, see the package-level AdvancedConfiguration example.
```
Comment (bogdandrutu) [inline config/service.go:77]: The "AdvancedConfiguration" is not in our "package". Please update the comment accordingly.

File: config/service.go [hunk 5]
```diff
@@ -55,6 +55,30 @@ type ServiceTelemetryLogs struct {
...
+	ErrorOutputPaths []string `mapstructure:"errorOutputPaths"`
+
+	// InitialFields is a collection of fields to add to the root logger.
+	InitialFields map[string]interface{} `mapstructure:"initialFields"`
```
Comment (bogdandrutu) [inline config/service.go:81]: We don't use "camel case" in our configuration, instead we use snake case. Everywhere.

**Review by bogdandrutu — CHANGES_REQUESTED** [review]

File: config/configunmarshaler/defaultunmarshaler.go [hunk 6]
```diff
@@ -168,9 +168,14 @@ func unmarshalService(srvRaw map[string]interface{}) (config.Service, error) {
 	srv := config.Service{
 		Telemetry: config.ServiceTelemetry{
 			Logs: config.ServiceTelemetryLogs{
-				Level:       zapcore.InfoLevel,
-				Development: false,
-				Encoding:    "console",
+				Level:             zapcore.InfoLevel,
```
Comment (bogdandrutu) [inline config/configunmarshaler/defaultunmarshaler.go:171]: Make sure these are similar with https://github.com/uber-go/zap/blob/master/config.go#L120 except Encoding which was "console" before and we don't want to break things in the PR.

Comment (pureklkl) [reply config/configunmarshaler/defaultunmarshaler.go:171]: Thanks for pointing it out, add a test to protect it, see TestDefaultLoggerConfig

File: config/service.go (hunk 1 above)
Comment (pureklkl) [reply config/service.go]: Sure, added it.

File: config/service.go (hunk 2 above)
Comment (pureklkl) [reply config/service.go]: The origin comments from Open is too long. I added a summary and point to the source file.

File: config/service.go (hunk 4 above)
Comment (pureklkl) [reply config/service.go]: Sure, removed it.

File: config/service.go (hunk 3 above)
Comment (pureklkl) [reply config/service.go]: Same changes with OutputPath.

File: config/service.go (hunk 5 above)
Comment (pureklkl) [reply config/service.go]: Sure, changed it to snake cases.

**Review by bogdandrutu — APPROVED** [review]

**Comment by bogdandrutu** [comment]
Please add a changelog entry

**Review by bogdandrutu — APPROVED** [review]

**Pull request merged**